src/main/java/uk/gov/justice/hmpps/prison/repository/ConditionalSqlService.kt

```diff
@@ -1,20 +1,24 @@
 package uk.gov.justice.hmpps.prison.repository
 
+import org.springframework.beans.factory.annotation.Value
 import org.springframework.context.annotation.Profile
 import org.springframework.stereotype.Service
 
 interface ConditionalSqlService {
-  fun getWaitClause(lockWaitTime: Int): String
+  fun getWaitClause(): String
+  fun getLockWaitTime(): Int
 }
 
 @Service
 @Profile("nomis")
-class OracleConditionalSqlService : ConditionalSqlService {
-  override fun getWaitClause(lockWaitTime: Int) = " WAIT $lockWaitTime"
+class OracleConditionalSqlService(@Value("\${lock.timeout:10}") private val lockWaitTimeSeconds: Int) : ConditionalSqlService {
+  override fun getWaitClause() = " WAIT $lockWaitTimeSeconds"
+  override fun getLockWaitTime() = lockWaitTimeSeconds
 }
 
 @Service
 @Profile("!nomis")
 class HsqlConditionalSqlService : ConditionalSqlService {
-  override fun getWaitClause(lockWaitTime: Int) = ""
+  override fun getWaitClause() = ""
+  override fun getLockWaitTime() = 0
```

Comment:
need conditional bean as hsql does not support the "for update wait n" select syntax